java/memory/src/test/java/org/apache/arrow/memory/TestBaseAllocator.java

```diff
@@ -478,6 +520,79 @@ public void testAllocator_overAllocateParent() throws Exception {
     }
   }
 
+  @Test
+  public void testAllocator_parentLimit_outcomeDetails() throws Exception {
+    try (final RootAllocator rootAllocator =
+        new RootAllocator(MAX_ALLOCATION)) {
+      try (final BufferAllocator childAllocator =
+          rootAllocator.newChildAllocator("child", 0, MAX_ALLOCATION / 2)) {
+        try (final BufferAllocator grandChildAllocator =
+            childAllocator.newChildAllocator("grandchild", MAX_ALLOCATION / 4, MAX_ALLOCATION)) {
+          try {
+            final ArrowBuf arrowBuf = grandChildAllocator.buffer(MAX_ALLOCATION);
```

Comment:
with JUnit5 you can use assertThrows instead of this try/catch patther.

Reply:
thanks, i didn't now this. fixed now.